scripts/gentest/src/main.rs

```diff
@@ -270,44 +272,50 @@ fn generate_assertions(ident: &str, node: &Value, use_rounding: bool) -> TokenSt
 
     let ident = Ident::new(ident, Span::call_site());
 
+    let gen_check = |expr: TokenStream, expected: f32| {
+        let cmp = if use_rounding {
+            quote!(layout.#expr != #expected)
+        } else {
+            quote!((layout.#expr - #expected).abs() >= 0.1)
+        };
+        quote!(
+            if #cmp {
+                mismatches += 1;
+                eprintln!("{:?}.{} mismatch: expected {} actual {}", #ident, stringify!(#expr), #expected, layout.#expr);
+            }
+        )
+    };
```

Comment:
This is very nice! At some point I am hoping to be able to print an actual diff between the expected and actual trees. But this is a much easier way of getting some of the benefit of that.